### PR TITLE
Uppercasing of properties in Output Template does not work. #395

### PR DESCRIPTION
**Pull request by the-dwyer**
Hi,

I added three unit tests for this bug as well as a possible fix.

Let me know what you think.

Thanks,

Michael

### CODE REVIEW

File: src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs [hunk 1]
```diff
@@ -82,11 +82,11 @@ public void Format(LogEvent logEvent, TextWriter output)
                 // rendering and support some additional formats: 'u' for uppercase
                 // and 'w' for lowercase.
                 var sv = propertyValue as ScalarValue;
-                if (sv != null && sv.Value is string)
+                if (sv != null)
                 {
                     var overridden = new Dictionary<string, LogEventPropertyValue>
                     {
-                        { pt.PropertyName, new LiteralStringValue((string) sv.Value) }
+                        { pt.PropertyName, new LiteralStringValue(sv.Value.ToString()) }
```
Comment (nblumhardt) [inline src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs]: Is it possible for `Value` to be null here?

**Comment by nblumhardt** [comment]
Thanks Michael! Added one note.

**Comment by the-dwyer** [comment]
Hi,

I added a unit test for the case when the value of the ScalarValue is null as well as the code to make the unit test pass.  I looked in the Render method of ScalarValue and it uses "null" when the value is null so that is what I used as well.

Let me know what you think.

Thanks,

Michael

File: src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs [hunk 2]
```diff
@@ -82,11 +82,12 @@ public void Format(LogEvent logEvent, TextWriter output)
                 // rendering and support some additional formats: 'u' for uppercase
                 // and 'w' for lowercase.
                 var sv = propertyValue as ScalarValue;
-                if (sv != null && sv.Value is string)
```
Comment (nblumhardt) [inline src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs]: I'm not sure I completely understand why this change works - previously we only used this rendering path for strings, but in the new version we'll end up rendering everything - integers, datetimes and so-on, down this path. Am I misunderstanding here? Thanks!

**Comment by nblumhardt** [comment]
Aaah- sorry, I just re-read https://github.com/serilog/serilog/issues/395 and realised what's going on here.

`:u` and `:w` are not valid on the level because it is an `enum`, not a `string`, and those specifiers only work for strings.

Can we please pause this PR and go back to the issue? I believe this case is already covered but will need to take another look at it.

**Comment by the-dwyer** [comment]
You bet!  I'll close this one and if I have another possible solution, I'll send in another pull request.

Michael
